File: backend/whatsapp-bot-service/src/redis/redis.ts
import Redis from 'ioredis';
import { config } from '../config';
import { logger } from '../utils/logger';

export const redis = new Redis(config.REDIS_URL);

redis.on('connect', () => {
  logger.info('Redis connected successfully to bot service client');
});

redis.on('error', (err) => {
  logger.error('Redis bot service connection error', err);
});

File: backend/crm-service/redis/redis.go
package redis

import (
	"context"

	"github.com/redis/go-redis/v9"
	"github.com/rs/zerolog/log"
)

func Connect(redisURL string) (*redis.Client, error) {
	opts, err := redis.ParseURL(redisURL)
	if err != nil {
		return nil, err
	}

	client := redis.NewClient(opts)
	if err := client.Ping(context.Background()).Err(); err != nil {
		return nil, err
	}

	log.Info().Msg("Redis connected")
	return client, nil
}
